app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-schemas.js

```diff
@@ -45,19 +49,60 @@
     }
   }
 
-  // get attributes from the data attribute to send to GA
+  // merge data attributes data into the event schema
   // only allow it if it already exists in the schema
   Schemas.prototype.mergeProperties = function (data, eventAttribute) {
     var schema = this.eventSchema()
     schema.event = eventAttribute
+    // exceptions should be inserted without checking for sub parameters
+    // exceptions should only come from code, not directly from data attributes
+    // e.g. link_path_parts is generated by JS and contains potentially non-unique
+    // sub parameter names i.e. '1', '2'
+    var exceptions = ['link_path_parts']
+
     for (var property in data) {
-      if (property in schema.event_data) {
-        schema.event_data[property] = data[property]
+      // some passed data might be undefined, don't want it to overwrite e.g. the index sub parameters
+      if (data[property] !== undefined) {
+        if (exceptions.indexOf(property) >= 0 || !this.isAnObject(data[property])) {
+          schema.event_data = this.addToObject(schema.event_data, property, data[property])
+        } else {
+          // we check for one level of nesting in the data attributes data
+          // this check can be remove once nesting is removed from all data attributes
```

Comment:
Possible typo - should `this check can be remove` be `this check can be removed`?

Reply:
Fixed, thanks.